spec/var-op-bash.test.sh

```diff
@@ -472,3 +458,81 @@ a
 A
 A
 ## END
+
+
+#### Array expansion with nullary var ops
+
+declare -a a=({1..9})
+declare -A A=(['a']=hello ['b']=world ['c']=osh ['d']=ysh)
+
+echo "@Q"
+argv.py "${a[@]@Q}"
+argv.py "${a[*]@Q}"
+argv.py "${A[@]@Q}"
+argv.py "${A[*]@Q}"
+argv.py "${u[@]@Q}"
+argv.py "${u[*]@Q}"
+
+echo "@P"
+argv.py "${a[@]@P}"
+argv.py "${a[*]@P}"
+argv.py "${A[@]@P}"
+argv.py "${A[*]@P}"
+argv.py "${u[@]@P}"
+argv.py "${u[*]@P}"
+
+echo "@a"
+argv.py "${a[@]@a}"
+argv.py "${a[*]@a}"
+argv.py "${A[@]@a}"
+argv.py "${A[*]@a}"
+argv.py "${u[@]@a}"
+argv.py "${u[*]@a}"
+
+## STDOUT:
+@Q
+['1', '2', '3', '4', '5', '6', '7', '8', '9']
+['1 2 3 4 5 6 7 8 9']
+['hello', 'world', 'osh', 'ysh']
+['hello world osh ysh']
+[]
+['']
+@P
+['1', '2', '3', '4', '5', '6', '7', '8', '9']
+['1 2 3 4 5 6 7 8 9']
+['hello', 'world', 'osh', 'ysh']
+['hello world osh ysh']
+[]
+['']
+@a
+['a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a']
+['a a a a a a a a a']
+['A', 'A', 'A', 'A']
+['A A A A']
+[]
+['']
+## END
+
+## OK bash STDOUT:
+@Q
+["'1'", "'2'", "'3'", "'4'", "'5'", "'6'", "'7'", "'8'", "'9'"]
```

Comment:
This is a big "OK bash" block, but our intention is to make it basically the same as bash

If I read it over, I can see the difference comes from 2 sources

1. quoting
2. ordering of assoc arrays


So in general I think we can try to

1. split up the test cases, e.g. for `[] ['']` which are identical
2. "adjust" the difference with sed, like we've already done

Then we will have more GREEN, which is easier to read/verify later IMO ... it matches our intention better


In some tests, I use `echo ${A[@]} | sort` and so forth to mask the difference in order

Reply:
#2217